pkg/scheduler/scheduler.go

```diff
@@ -575,6 +575,10 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 		return
 	}
 
+	// In some unit tests, <sched.SchedulingQueue> can be nil.
+	if sched.SchedulingQueue != nil {
+		defer sched.SchedulingQueue.ClearProcessingPod(pod.UID)
```

Comment:
🤔 Shouldn't this be conditional to being unschedulable? Otherwise, it should be done in the binding routine.

Reply:
Ah, that's true. The logic should be updated:
- if the logic returns during the scheduling cycle, the above `defer()` should run
- otherwise, we should signal to abort the above `deer()` (in the scheduling cycle); and run it in the the binding cycle.

Let me update the code.